packages/react/src/components/Tree/Tree.tsx

```diff
@@ -146,9 +146,35 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
   }
 
   static getAutoControlledStateFromProps(nextProps: TreeProps, prevState: TreeState) {
-    const itemsForRender = Tree.getItemsForRender(nextProps.items)
+    const { items } = nextProps
+    const itemsForRender = Tree.getItemsForRender(items)
+    let { activeItemIds } = nextProps
+
+    if (!activeItemIds && items) {
+      activeItemIds = prevState.activeItemIds
+
+      const expandedItemsGenerator = items =>
+        _.reduce(
+          items,
+          (acc, item) => {
+            if (item['expanded'] && acc.indexOf(item['id']) === -1) {
```

Comment:
The items could be defined as render functions too, we should handle that case, or at least check whether it is an object.

Reply:
Currently it won't work well with render functions. We are calculating some props that we pass to TreeItem from Tree in `getItemsForRender`. However we probably can support this scenario with render prop as well. We just call the children function with the props from `itemsForRender` so the child component has access to them as well.

I would consider working on this in a separate PR. And just perform here the object check as per suggestion.

What do you think @mnajdova @jurokapsiar @miroslavstastny ?